### PR TITLE
.coafile: Add .coafile

### PR DESCRIPTION
**Pull request by frextrite**
"--show-bears --filter-by language Java" bears
are used for analyzing all Java files
present in the project

### CODE REVIEW

**Comment by ksdme** [comment]
- No newline at the end of file
- `bears = ArtisticStyleBear, CheckstyleBear, coalaBear, CPDBear, DocGrammarBear, DocumentationStyleBear, DuplicateFileBear,
FilenameBear, HTTPSBear, InferBear, InvalidLinkBear, JavaPMDBear, KeywordBear, LicenseCheckBear, LicenseHeaderBear,
LineCountBear, LineLengthBear, MementoBear, SpaceConsistencyBear, URLBear, URLHeadBear
` Too many bears in a single section. Please divide them into more maintainable form such as a section of Documentation another for misc such as Licenses etc

**Comment by frextrite** [comment]
I just added all the bears listed as compatible with `Java`. Didn't know what half of them really do. I will segregate in proper sections them.

**Comment by frextrite** [comment]
Split up into multiple sections. Ready to merge :rocket:

File: .coafile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[java]
+tags = jetbrains
+bears = ArtisticStyleBear, CheckstyleBear, coalaBear, CPDBear, JavaPMDBear
+files = **.java
+language = Java
+checkstyle_configs = google
+use_spaces = true
+indent_size = 4
+
+[java.misc]
```
Comment (ksdme) [inline .coafile]: Could just be misc? I suppose you have this to inherit the `files` parameter? Please use a generic section, you need to check the whole of files for duplicates etc, not just Java files. Consider a section named `all`.

File: .coafile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+bears = DuplicateFileBear, FilenameBear, SpaceConsistencyBear, LineLengthBear
+file_naming_convention = pascal
+max_line_length = 120
+
+[java.network]
+enable = 0
```
Comment (ksdme) [inline .coafile]:
```suggestion
enable = false
```

File: .coafile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+bears = HTTPSBear, InvalidLinkBear
+
+[java.documentation]
+enable = 0
+bears = DocGrammarBear
+
+[java.license]
+enable = 0
```
Comment (ksdme) [inline .coafile]: Why is this section disabled?

File: .coafile [hunk 4]
```diff
@@ -0,0 +1,25 @@
+[java]
+tags = jetbrains
```
Comment (ksdme) [inline .coafile]: :+1:

File: .coafile [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+checkstyle_configs = google
+use_spaces = true
+indent_size = 4
+
+[java.misc]
+bears = DuplicateFileBear, FilenameBear, SpaceConsistencyBear, LineLengthBear
+file_naming_convention = pascal
+max_line_length = 120
+
+[java.network]
```
Comment (ksdme) [inline .coafile]: `all.network`. You get the idea.

**Comment by ksdme** [comment]
We also need to take care of https://github.com/frextrite/coala-jetbrains/issues/4

File: .coafile (hunk 1 above)
Comment (frextrite) [reply .coafile]: On it.

File: .coafile (hunk 3 above)
Comment (frextrite) [reply .coafile]: This section is not required during current development but may be required after.

File: .coafile (hunk 5 above)
Comment (frextrite) [reply .coafile]: :+1:

File: .coafile (hunk 2 above)
Comment (frextrite) [reply .coafile]: Will change this aswell.

**Pull request merged**